server/src/main/java/io/druid/server/coordinator/DruidCluster.java

```diff
@@ -113,9 +115,11 @@ private void addHistorical(ServerHolder serverHolder)
 
   public Collection<ServerHolder> getAllServers()
   {
-    return historicals.values().stream()
-                      .flatMap(Collection::stream)
-                      .collect(() -> realtimes, Set::add, Set::addAll);
+    final List<ServerHolder> allServers = historicals.values().stream()
```

Comment:
@jihoonson could it be turned into a proper immutable then, to avoid bugs? Also getAllServers() could be cached.